### PR TITLE
feat(chat-e2e): implemented export replay with attachments tests

### PR DESCRIPTION
**Pull request by irinakartun**
**Description:**

implemented export replay with attachments tests

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1894.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat-e2e/src/testData/conversationHistory/conversationData.ts [hunk 1]
```diff
@@ -215,17 +215,18 @@ export class ConversationData extends FolderData {
...
       replayIndex,
     );
+    const conversationCopy = JSON.parse(
+      JSON.stringify(conversation),
+    ) as Conversation;
     const partialAssistantMessage = replayIndex
-      ? conversation.messages[replayIndex + 2]
-      : conversation.messages.findLast((m) => m.role === 'assistant');
+      ? conversationCopy.messages[replayIndex + 2]
+      : conversationCopy.messages.findLast((m) => m.role === 'assistant');
```
Comment (nartovm) [inline apps/chat-e2e/src/testData/conversationHistory/conversationData.ts]: I know that this is not what you have changed, but why do we have +2 index here? Probably we need some comments?

File: apps/chat-e2e/src/testData/conversationHistory/conversationData.ts [hunk 2]
```diff
@@ -215,17 +215,18 @@ export class ConversationData extends FolderData {
       conversation,
       replayIndex,
     );
+    const conversationCopy = JSON.parse(
```
Comment (nartovm) [inline apps/chat-e2e/src/testData/conversationHistory/conversationData.ts]: Can we somehow omit the creation of the deep copy here? Why do we need it here?

File: apps/chat-e2e/src/testData/conversationHistory/conversationData.ts [hunk 3]
```diff
@@ -215,17 +215,18 @@ export class ConversationData extends FolderData {
...
+      JSON.stringify(conversation),
+    ) as Conversation;
     const partialAssistantMessage = replayIndex
-      ? conversation.messages[replayIndex + 2]
-      : conversation.messages.findLast((m) => m.role === 'assistant');
+      ? conversationCopy.messages[replayIndex + 2]
```
Comment (nartovm) [inline apps/chat-e2e/src/testData/conversationHistory/conversationData.ts]: Consider adding validation for replayIndex to ensure it's within the bounds of the messages array

File: apps/chat-e2e/src/tests/chatExportImportWithAttachment.test.ts [hunk 4]
```diff
@@ -502,3 +509,284 @@ dialTest(
...
+    setTestIds('EPMRTC-914');
+    let geminiProVisionConversation: Conversation;
+    let gptProVisionConversation: Conversation;
+    let historyConversation: Conversation;
+    let replayConversation: Conversation;
+    let exportedData: UploadDownloadData;
+
+    await dialTest.step(
+      'Prepare conversation with image, pdf attachments in the requests and replay conversation based on it',
+      async () => {
```
Comment (nartovm) [inline apps/chat-e2e/src/tests/chatExportImportWithAttachment.test.ts:541]: The code for preparing conversations with attachments is repeated for both models. Probably we can refactor it into a separate function

File: apps/chat-e2e/src/tests/chatExportImportWithAttachment.test.ts [hunk 5]
```diff
@@ -502,3 +509,284 @@ dialTest(
...
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Verify request attachments can be opened and downloaded',
+      async () => {
+        geminiProVisionAttachmentPath = `${API.importFileRootPath(BucketUtil.getBucket())}/${Attachment.sunImageName}`;
+        await chatMessages.expandChatMessageAttachment(
+          1,
```
Comment (nartovm) [inline apps/chat-e2e/src/tests/chatExportImportWithAttachment.test.ts]: Do we consider hardcoded indexes a good approach?

Comment (irinakartun) [reply apps/chat-e2e/src/tests/chatExportImportWithAttachment.test.ts]: fixed

File: apps/chat-e2e/src/tests/chatExportImportWithAttachment.test.ts (hunk 4 above)
Comment (irinakartun) [reply apps/chat-e2e/src/tests/chatExportImportWithAttachment.test.ts:541]: done

File: apps/chat-e2e/src/testData/conversationHistory/conversationData.ts (hunk 3 above)
Comment (irinakartun) [reply apps/chat-e2e/src/testData/conversationHistory/conversationData.ts]: added validation into `fillReplayData()` method

File: apps/chat-e2e/src/testData/conversationHistory/conversationData.ts (hunk 2 above)
Comment (irinakartun) [reply apps/chat-e2e/src/testData/conversationHistory/conversationData.ts]: `conversation.messages` and `conversation.replay.replayUserMessagesStack` objects cannot refer to the same object that's why a copy is required

File: apps/chat-e2e/src/testData/conversationHistory/conversationData.ts (hunk 1 above)
Comment (irinakartun) [reply apps/chat-e2e/src/testData/conversationHistory/conversationData.ts]: left comments

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1894.nightly-test.deltixhub.io
>E2E tests status: success

**Review by nartovm — APPROVED** [review]

**Pull request merged**